### PR TITLE
Update react/react-dom to 16.13.0 and docgen to 5.3.0 [ref dash#1531]

### PR DESCRIPTION
**Pull request by AnnMarieW**
This pull request is for updating the docstring comments so they will be compatible with the new style docstrings in [dash #1531](https://github.com/plotly/dash/pull/153)

This project started with   https://github.com/plotly/dash/issues/1205

You can see results for other libraries in dash-doc #https://github.com/plotly/dash-docs/pull/1063

This repo will need to be rebuilt with a more recent version of docgen than the one currently specified (2.21.0).  I rebuilt the library with docgen 5.3.0, react 16.13.1 and react-dom 16.13.1.  I see that  https://github.com/plotly/dash-cytoscape/pull/81 is still open, so I don't know if these version changes will cause any problems.   The rebuilt library passed all tests.

Here are some before and after screenshots for dash-cytoscape in dash-docs and in the python console.

Before:
-----
-----

cyto1

------------
--------

After

cyto3

-----
-----
In the console: Before:
------
cyto_console1

In the console: After:
-----
-----
cyto_console2

-

### CODE REVIEW

**Comment by xhlulu** [comment]
@AnnMarieW Thank you for opening the PR, the updated docstring looks much better! And thank you for the thorough and well-written PR description.

@alexcjohnson @Marc-Andre-Rivet I noticed that the dash version in `requirements.txt` is fairly outdated; do you think this could affect the build process wrt the `react` and `react-dom` version?

**Comment by AnnMarieW** [comment]
@xhlulu I'm glad you like it!  I forgot to mention that I also used the latest version of dash to rebuild and run the tests.  The main issue I ran into was that docgen 2.21.0 didn't recognized PropTypes.exact so it needed to be upgraded.  I just bumped it to a version being used in dash-table

**Comment by alexcjohnson** [comment]
hah yes, 0.39 is stretching the definition of "fairly" outdated at this point. But what is `requirements.txt` even used for? `setup.py` only lists `dash` (and with no version). There's also `tests/requirements.txt` listing an extremely old dash (but with `>=` so it should be getting the latest) as well as `pytest-dash` - would be great at some point to transition this to `dash[testing]` but that will require rewriting some of the tests themselves.

**Comment by xhlulu** [comment]
@alexcjohnson Thanks for mentioning this. `requirements.txt` would ideally be the requirements needed to run the demos locally and developing the libraries. Perhaps it's time to delete `test/requirements.txt` and move everything to `requirements.txt`, so we won't have to track/manage two requirements files?

I've also created [a project](https://github.com/plotly/dash-cytoscape/projects/4#card-53220088) to track everything relating to tests, as well as opened an issue to track the refactoring of tests.

**Review by xhlulu — COMMENTED** [review]
I've added the build here in this test branch: https://github.com/plotly/dash-cytoscape/tree/update-docgen

I don't really have anything to say if the tests pass. @alexcjohnson is there anything else before :woman_dancing: ?

**Comment by alexcjohnson** [comment]
This looks great to me. @AnnMarieW let's get https://github.com/plotly/dash-docs/pull/1063 merged (WITHOUT any updates to the component packages) before we merge any of the updates to the component packages themselves.

**Comment by AnnMarieW** [comment]
Sounds good.  And  https://github.com/plotly/dash/pull/1531 will also need to be merged before updating the component packages?

File: src/lib/components/Cytoscape.react.js
Comment (xhlulu) [inline src/lib/components/Cytoscape.react.js:518]: Do we want to use `exact` or would `shape` be nicer here? I guess it doesn't hurt to warn the user about unwanted props but in this case I don't think the original JS library intended to have a strict typing here. cc @alexcjohnson

Comment (AnnMarieW) [reply src/lib/components/Cytoscape.react.js:518]: I agree, shape would be nicer if it's acceptable to include other props.   I did that in the `layout` prop where it was clear from your description that that was the case

Comment (alexcjohnson) [reply src/lib/components/Cytoscape.react.js:518]: Ah, good catch! In cytoscape you can add random extra data to elements and you can reference that data for example in stylesheets. So `shape` actually would be preferable here.

Comment (xhlulu) [reply src/lib/components/Cytoscape.react.js:518]:
```suggestion
            PropTypes.shape({
```

File: src/lib/components/Cytoscape.react.js
Comment (xhlulu) [inline src/lib/components/Cytoscape.react.js:524]:
```suggestion
                data: PropTypes.shape({
```

File: src/lib/components/Cytoscape.react.js
Comment (xhlulu) [inline src/lib/components/Cytoscape.react.js:539]:
```suggestion
                position: PropTypes.shape({
```

File: package.json
Comment (xhlulu) [inline package.json:43]: I wonder if it's better to fix it to 16.13.0 since that's what's done on `plotly/dash-core-components`. Not sure if it might break dash if react/react-dom are upgraded beyond that.
```suggestion
    "react": "16.13.0",
```

File: package.json
Comment (xhlulu) [inline package.json:45]: Same as above
```suggestion
    "react-dom": "16.13.0"
```

File: package.json
Comment (xhlulu) [inline package.json:60]: Wonder if a `~` would be better than `^` here (I just learned the differences [here](https://stackoverflow.com/questions/22343224/whats-the-difference-between-tilde-and-caret-in-package-json))

**Review by xhlulu — CHANGES_REQUESTED** [review]
@AnnMarieW Sorry for getting back to you so late, I didn't realize dash#1531 was merged! I suggested a few changes and I'll merge it once I figure out what's wrong with the percy tests!

**Comment by AnnMarieW** [comment]
@xhlulu Thanks for the suggested changes.  I approved the changes from `exact` to `shape` and will leave the rest open for you or the other reviewers.

**Comment by xhlulu** [comment]
@AnnMarieW Thank you!

@alexcjohnson I'm happy to merge it as is and I can figure out the right package.json dependencies and building in another PR

**Review by xhlulu — APPROVED** [review]
💃

**Review by alexcjohnson — APPROVED** [review]
💃

**Pull request merged**
